README.md

```diff
@@ -1,215 +1,224 @@
-
-The actual code for generating these classes is in the
-[`srcgen` folder](https://github.com/MingweiSamuel/Camille/tree/release/3.x.x/srcgen).
-The C#-generating code is in `*.cs.dt` files and is written in NodeJS, using
-[doT.js templates](https://olado.github.io/doT/index.html).
+# Camille
+
+[![Github Actions](https://img.shields.io/github/workflow/status/MingweiSamuel/Camille/CI/release/3.x.x?label=release/3.x.x&logo=github&style=flat-square)](https://github.com/MingweiSamuel/Camille/actions?query=workflow%3ACI+branch%3Arelease%2F3.x.x) [![NuGet Stable](https://img.shields.io/nuget/v/Camille.RiotGames.svg?style=flat-square)](https://www.nuget.org/packages/Camille.RiotGames/) [![NuGet Pre Release](https://img.shields.io/nuget/vpre/Camille.RiotGames.svg?style=flat-square)](https://www.nuget.org/packages/Camille.RiotGames/absoluteLatest) [![API Reference](https://img.shields.io/badge/docfx-Camille-brightgreen.svg?style=flat-square)](http://www.mingweisamuel.com/Camille/)
+
+C# Library for the [Riot Games API](https://developer.riotgames.com/)
+
+Camille's goals are _speed_, _reliability_, and _maintainability_. Camille handles rate limits and large requests with ease.
+Data classes are automatically generated from the
+[Riot API Reference](https://developer.riotgames.com/api-methods/) ([Swagger](http://www.mingweisamuel.com/riotapi-schema/tool/)).
+
+## Features
+
+* Fast, asynchronous, thread-safe.
+* Automatically retries failed requests.
+* Automatic up-to-date nightlies, reflecting Riot API changes within 24 hours.
+* Multi-targeted: .NET Standard 1.1+, .NET Framework 4.5+, .NET Core 2.0, .NET Standard 2.1+, .NET Core 3.0+.
+* Highly-configurable.
+* Riot API LOL-V4, TFT, LOR support.
+* C# 8 nullable support.
+
+## Installation
+
+[Install via NuGet](https://www.nuget.org/packages?sortby=created-desc&q=Camille&prerel=True) (`Camille.RiotGames`). 
+
+Type the following in the package manager console:
+
+    Install-Package Camille.RiotGames -IncludePrerelease
+    
+Or use the .NET CLI:
+
+    dotnet add package Camille.RiotGames --prerelease
+
+## Usage
+
+All API interactions are done using a `RiotApi` instance.
+`RiotApi.NewInstance` takes either just an API key (for default settings) or a `IRiotApiConfig` instance (for custom settings).
+
+```c#
+using Camille.Enums;
+using Camille.RiotGames;
+```
+
+```c#
+var riotApi = RiotGamesApi.NewInstance("RGAPI-12345678-abcd-1234-abcd-123456abcdef");
+// OR
+var riotApi = RiotGamesApi.NewInstance(
+    new RiotGamesApiConfig.Builder("RGAPI-12345678-abcd-1234-abcd-123456abcdef")
+    {
+        MaxConcurrentRequests = 200,
+        Retries = 10,
+        // ...
+    }.Build()
+);
+// OR with a proxy server
+var riotApi = RiotGamesApi.NewInstance(
+    new RiotGamesApiConfig.Builder("")
+    {
+        // ...
+        ApiURL = "proxy.myexample.app",
+    }.Build()
+);
+// Which will send requests like https://proxy.myexample.app/lol/summoner/v4/summoners/by-name/NAME?region=NA1
+```
```

Comment:
Doesn't really need to be top of README, should be pretty obvious what it does from the docs